code/script.js

```diff
@@ -1,51 +1,170 @@
+const nameInput = document.getElementById("name-input");
+const form = document.getElementById("name-form");
+const sound = document.getElementById("clickSound");
 
-// If you need any global variables that you can use across different functions, declare them here:
+// Step 1 and Step 2
+// When website loaded, chatbot asks first question.
+// This function will add a chat bubble in the correct place based on who the sender is
+const showMessage = (msg, sender) => {
+  console.log("MESSAGE IS:", msg);
+  console.log("SENDER IS:", sender);
+  if (sender === "bot") {
+    chat.innerHTML += `<section class="bot-msg">
+    <img src="assets/bot.svg" alt="chat bot"/>
+    <div class="bubble bot-bubble">
+      <p>${msg}</p>
+    </div>
+    </section>
+    `;
+  } else if (sender === "user") {
+    sound.play();
+    chat.innerHTML += `<section class="user-msg">
+  <div class="bubble user-bubble">
+    <p>${msg}</p>
+  </div>
+  <img src="assets/user.svg" alt="user bot"/>
+  </section>`;
+  }
+  // This little thing makes the chat scroll to the last message when there are too many to be shown in the chat box
+  chat.scrollTop = chat.scrollHeight;
+};
+const greetUser = () => {
+  showMessage(`Hello there, What's your name?`, `bot`);
+  // show message.the message should be from the bot
+};
+setTimeout(greetUser, 1000);
+// greetUser()
 
+// Step 3
+// when user click the button, then show user's message
+const handleNameinput = (event) => {
+  event.preventDefault();
+  const userName = nameInput.value;
+  showMessage(`My name is ${userName}`, `user`);
+  nameInput.value = ""; //clear nameInput's value
+  setTimeout(() => askFoodType(userName), 1500);
+};
 
-// Declare your functions after this comment
+btn.addEventListener("click", handleNameinput);
 
-// This function will add a chat bubble in the correct place based on who the sender is
-const showMessage = (message, sender) => {
-  // the if statement checks if the sender is 'user' and if that's the case it inserts an html senction inside the chat with the posted message
-  if (sender === 'user') {
-    chat.innerHTML += `
-      <section class="user-msg">
-        <div class="bubble user-bubble">
-          <p>${message}</p>
-        </div>
-        <img src="assets/user.png" alt="User" />  
-      </section>
-    `
-    // the else if statement checks if the sender is a bot and if that's the case it inserts an html senction inside the chat with the posted message
-  } else if (sender === 'bot') {
-    chat.innerHTML += `
-      <section class="bot-msg">
-        <img src="assets/bot.png" alt="Bot" />
-        <div class="bubble bot-bubble">
-          <p>${message}</p>
-        </div>
-      </section>
-    `
+// bot ask user: what type of food would like to order and create menu button
+const askFoodType = (userName) => {
+  showMessage(
+    `Nice to meet you ${userName}! 😁 What type of food would you like to order?`,
+    `bot`
+  );
+  form.innerHTML = `
```

Comment:
I understand that `form.innerHTML = `... ` `  instead of `form.innerHTML += ` ...` ` prompts the content inside the local scope and disappears with it when the event is caught (end of code block running)?  Clean coding.